src/docfx/lib/schema/JsonSchemaValidator.cs

```diff
@@ -553,6 +553,25 @@ private void PostValidateDocsetUnique(List<(string name, Error)> errors)
                         }
                     }
                 }
+                else
+                {
+                    if (string.IsNullOrEmpty(dependentFieldNameWithIndex))
```

Comment:
may be refactor this chunk of code in to a separate method? This seems dup with ln 538 ~  552

Reply:
not exactly same. Refactored code to reflect this. And if extract code to another method, that would result in a long parameters list for the method..